src/Features/VisualBasic/Portable/GoToDefinition/VisualBasicGoToDefinitionSymbolService.vb

```diff
@@ -20,5 +21,141 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.GoToDefinition
         End Function
+
+        Protected Overrides Function GetTargetPositionIfControlFlow(token As SyntaxToken) As Integer?
+            Dim node = token.GetRequiredParent()
+
+            If token.IsKind(SyntaxKind.ReturnKeyword, SyntaxKind.YieldKeyword) Then
+                Return FindContainingReturnableConstruct(node).GetFirstToken().Span.Start
+            End If
+
+            Dim continueTarget = TryGetContinueTarget(node)
+            If continueTarget IsNot Nothing Then
+                Return continueTarget.GetFirstToken().Span.Start
+            End If
+
+            Dim exitTarget = TryGetExitTarget(node)
+            If exitTarget IsNot Nothing Then
+                Select Case node.Kind()
+                    Case SyntaxKind.ExitSubStatement
+                    Case SyntaxKind.ExitFunctionStatement
+                    Case SyntaxKind.ExitPropertyStatement
+                        Return exitTarget.GetFirstToken().Span.Start
+                End Select
+
+                Return exitTarget.GetLastToken().Span.End
+            End If
+
+            Return Nothing
+        End Function
+
+        Private Shared Function TryGetExitTarget(node As SyntaxNode) As SyntaxNode
+            Select Case node.Kind()
+                Case SyntaxKind.ExitSelectStatement
+                    Return FindContainingSelect(node)
+                Case SyntaxKind.ExitWhileStatement
+                    Return FindContainingWhile(node)
+                Case SyntaxKind.ExitForStatement
+                    Return FindContainingFor(node)
+                Case SyntaxKind.ExitDoStatement
+                    Return FindContainingDoLoop(node)
+                Case SyntaxKind.ExitTryStatement
+                    Return FindContainingTry(node)
+                Case SyntaxKind.ExitPropertyStatement
+                    Return FindContainingReturnableConstruct(node)
+                Case SyntaxKind.ExitSubStatement
+                    Return FindContainingReturnableConstruct(node)
+                Case SyntaxKind.ExitFunctionStatement
+                    Return FindContainingReturnableConstruct(node)
+            End Select
+
+            Return Nothing
+        End Function
+
+        Private Shared Function TryGetContinueTarget(node As SyntaxNode) As SyntaxNode
+            Select Case node.Kind()
+                Case SyntaxKind.ContinueWhileStatement
+                    Return FindContainingWhile(node)
+                Case SyntaxKind.ContinueForStatement
+                    Return FindContainingFor(node)
+                Case SyntaxKind.ContinueDoStatement
+                    Return FindContainingDoLoop(node)
+            End Select
+
+            Return Nothing
+        End Function
+
+        Private Shared Function FindContainingSelect(node As SyntaxNode) As SyntaxNode
+            While node IsNot Nothing AndAlso Not node.IsKind(SyntaxKind.SelectBlock)
+                node = node.Parent
+
+                If node.IsReturnableConstruct() Then
```

Comment:
feels like  we can remove this.

Reply:
I think it's safer to check that we don't accidentally cross method/lambda/etc boundaries. I'm not sure how we parse `Exit Select` when we're not in a `Select`, but I'm assuming that we may be parsing it correctly as `Exit Select`.
Same comment applies for other such boundary checks.